google/cloud/internal/async_retry_loop_test.cc

```diff
@@ -55,46 +58,66 @@ std::unique_ptr<BackoffPolicy> TestBackoffPolicy() {
       .clone();
 }
 
+using TimerResult = StatusOr<std::chrono::system_clock::time_point>;
+
 class AsyncRetryLoopCancelTest : public ::testing::Test {
  protected:
-  int CancelCount() {
-    std::lock_guard<std::mutex> lk(mu_);
-    int n = cancel_count_;
-    cancel_count_ = 0;
-    return n;
-  }
+  std::size_t CancelCount() { return sequencer_.CancelCount(); }
 
   future<StatusOr<int>> SimulateRequest(int x) {
-    promise<Status> p([this] {
-      std::lock_guard<std::mutex> lk(mu_);
-      ++cancel_count_;
-    });
-    auto f = p.get_future().then([x](future<Status> g) -> StatusOr<int> {
-      auto status = g.get();
-      if (!status.ok()) return status;
-      return 2 * x;
-    });
-    std::lock_guard<std::mutex> lk(mu_);
-    requests_.push_back(std::move(p));
-    cv_.notify_one();
-    return f;
+    return sequencer_.PushBack("Request").then(
+        [x](future<Status> g) -> StatusOr<int> {
+          auto status = g.get();
+          if (!status.ok()) return status;
+          return 2 * x;
+        });
+  }
+
+  future<TimerResult> SimulateDeadlineTimer(
+      std::chrono::system_clock::time_point tp) {
+    return sequencer_.PushBack("DeadlineTimer")
+        .then([tp](future<Status> g) -> TimerResult {
+          auto status = g.get();
+          if (!status.ok()) return status;
+          return tp;
+        });
+  }
+
```

Comment:
Should have pointed this out in the previous review, but this function is unused.